src/main/java/com/cgm/infolab/controller/ChatController.java

```diff
@@ -54,37 +53,93 @@ public void welcome(Authentication principal){
+            return messageDto;
+        } else {
+            return null;
+        }
+    }
+
+    @MessageMapping("/chat.edit")
+    @SendTo("/topic/public")
+    public WebSocketMessageDto editMessage(@Payload WebSocketMessageDto messageDto, Principal principal) {
+        int rowsAffected = chatService.editMessageById(Username.of(principal.getName()), messageDto.getEdit().getId(), messageDto.getEdit().getContent());
+        if (rowsAffected == 1) {
+            return messageDto;
+        } else {
+            return null;
+        }
+    }
+    //endregion
+
+    //region PRIVATE ROOMS
     @MessageMapping("/chat.send.{destinationUser}")
     @SendTo("/queue/{destinationUser}")
     @SendToUser("/topic/me")
-    ChatMessageDto sendMessageToUser(
-            @Payload ChatMessageDto message,
+    public WebSocketMessageDto sendMessageToUser(
+            @Payload WebSocketMessageDto message,
             @DestinationVariable String destinationUser,
             SimpMessageHeaderAccessor headerAccessor,
             Principal principal){
         String username = (String) headerAccessor.getSessionAttributes().get("username");
         log.info(String.format("message from %s to %s", username, destinationUser));
         ChatMessageEntity messageEntity = chatService.saveMessageInDb(
-                message, Username.of(principal.getName()),
-                RoomName.of(Username.of(principal.getName()),
-                Username.of(destinationUser)),
+                message.getChat(),
+                Username.of(principal.getName()),
+                RoomName.of(Username.of(principal.getName()), Username.of(destinationUser)),
                 Username.of(destinationUser)
         );
-        return FromEntitiesToDtosMapper.fromEntityToChatMessageDto(messageEntity);
+        return WebSocketMessageDto.ofChat(FromEntitiesToDtosMapper.fromEntityToChatMessageDto(messageEntity));
+    }
+
+    @MessageMapping("/chat.delete.{destinationUser}")
+    @SendTo("/queue/{destinationUser}")
+    @SendToUser("/topic/me")
+    public WebSocketMessageDto deleteMessagePrivateRoom(
+            @Payload WebSocketMessageDto message,
+            @DestinationVariable String destinationUser,
+            Principal principal){
+        int rowsAffected = chatService.deleteMessageById(Username.of(principal.getName()), message.getDelete().getId());
+        if (rowsAffected == 1) {
+            return message;
+        } else {
+            return null;
+        }
+    }
+
+    @MessageMapping("/chat.edit.{destinationUser}")
+    @SendTo("/queue/{destinationUser}")
+    @SendToUser("/topic/me")
+    public WebSocketMessageDto editMessagePrivateRoom(
+            @Payload WebSocketMessageDto message,
+            @DestinationVariable String destinationUser,
+            Principal principal){
+        int rowsAffected = chatService.editMessageById(Username.of(principal.getName()), message.getEdit().getId(), message.getEdit().getContent());
```

Comment:
Forse dovremmo gestire le casistiche di errore. Se `message.getEdit` fosse null, dovremmo come minimo, esplicitamente ignorarlo. Oppure rimettere un messaggio di errore al mittente.

Supponendo malizia da parte del mittente, forse in questo caso sarebbe meglio ignorare e loggare livello `WARN` la ricezione di una richiesta non conforme.